### ISSUE TITLE
save() flexible put/post decision

### ISSUE THREAD

**Issue by hesalx** [mgonto/restangular#832]
`save()` decides to POST or to PUT depending on the `fromServer` field.
But I've met the situation when this doesn't fit.

tl;dr — ID field can be reliable to determine create or update is needed.

There is the known REST-related route '/new'. Usually it returns HTML with a blank form to create a new resource.
But I've found it to be very useful for json API too. I can return some template object with needed field filled with defaults and without ID. It's also convenient whenever my model has dynamic fields, which behaves like a part of the model, not as an associated resource.
Yes, I could retrieve a 'new' object and fill a fresh restangular instance with it's values, but it will not be as natural as could be. And I should know a list of properties to copy, which I don't in case of dynamic fields.

So when I want to make a creating page I retrieve an object from the server with `/new`. But `save()` does PUT and breaks the things because there's no ID. And I have to always check `if resource.id` to put or post. 
RESTful is cool, but not almighty. Server doesn't have to give you only persisted resources, eventually :)
It would be great to have an ability to configure restangular for this.
